_pylibczi/_pylibczi.cpp

```diff
@@ -207,6 +228,147 @@ static PyObject *cziread_allsubblocks(PyObject *self, PyObject *args) {
+    };
+
+    int ret_val = 1;
+     std::for_each(tbl.begin(), tbl.end(), [&](std::pair< const std::string, libCZI::DimensionIndex > &kv)->void{
+         PyObject *pyInt = PyDict_GetItemString(obj, kv.first.c_str() );
+         if(pyInt != NULL){
+             dims->Set(kv.second, static_cast<int>(PyLong_AsLong(pyInt)));
+             if( PyErr_Occurred() != NULL) {
+                 PyErr_SetString(PylibcziError,
+                                 "problem converting Dictionary of dims, should be C=1 meaning Dimension = Integer");
+                 ret_val = 0;
+             }
+         }
+     });
+     return ret_val; // success
+ }
+
+static int listToIntRect(PyObject *obj, void * rect_p){
+    libCZI::IntRect *rect = static_cast<libCZI::IntRect *>(rect_p);
+    if( obj == Py_None ){
+        rect->x = rect->y = rect->w = rect->h = -1;
+        return 1;
+    }
+
+    if(!PyTuple_Check(obj)){
+        PyErr_SetString(PylibcziError, "region not set to a tuple, please use correct syntax");
+        return 0;
+    }
+
+    if(PyTuple_Size(obj) != 4){
+        PyErr_SetString(PylibcziError,
+                        "(Error) Syntax: region=(x, y, w, h) w/ x y w h all integers.");
+        return 0;
+    }
+
+    rect->x = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 0)));
+    rect->y = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 1)));
+    rect->w = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 2)));
+    rect->h = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 3)));
+    if( PyErr_Occurred() != NULL) { // collectively check if an PyErr occurred
+        PyErr_SetString(PylibcziError, "problem converting region=(x: [Int], y: [Int], w: [Int], h: [Int])");
+        return 0;
+    }
+    return 1;
+}
+
+bool isValidRegion(const libCZI::IntRect &inBox, const libCZI::IntRect &cziBox ){
+    bool ans = true;
+    // check origin is in domain
+    if( inBox.x < cziBox.x || cziBox.x + cziBox.w < inBox.x) ans = false;
+    if( inBox.y < cziBox.y || cziBox.y + cziBox.h < inBox.y) ans = false;
+
+    // check  (x1, y1) point is in domain
+    int x1 = inBox.x + inBox.w;
+    int y1 = inBox.y + inBox.h;
+    if( x1 < cziBox.x || cziBox.x + cziBox.w < x1) ans = false;
+    if( y1 < cziBox.y || cziBox.y + cziBox.h < y1) ans = false;
+
+    if(!ans) {
+        std::cerr << "Conflict region requested("
+                  << inBox.x << ", " << inBox.y << ", " << inBox.w << ", " << inBox.h
+                  << ") in plane ("
+                  << cziBox.x << ", " << cziBox.y << ", " << cziBox.w << ", " << cziBox.h << ")!"
+                  << std::endl;
+    }
+
+    if(inBox.w < 1 || 1 > inBox.h){
+        std::cerr << "Ill-defined region, with and heigh must be positive values! ("
+                << inBox.x << ", " << inBox.y << ", " << inBox.w << ", " << inBox.h << ")" << std::endl;
+        ans = false;
+    }
+
+    return ans;
+}
+
+static PyObject *cziread_mosaic(PyObject *self, PyObject *args) {
```

Comment:
A comment here explaining how plane reading is different than mosaic reading, unless I am mistaken? Iirc, plane reading is now how you read the image dims, instead of using metadata, and mosaic reading is just a newly supported CZI image format.